modules/@apostrophecms/asset/index.js

```diff
@@ -286,15 +373,43 @@ module.exports = {
             return output;
           }
 
-          function merge() {
+          function merge(scene) {
+            const jsModules = `${scene}-module-bundle.js`;
+            const jsNoModules = `${scene}-nomodule-bundle.js`;
+            const css = `${scene}-bundle.css`;
+            fs.writeFileSync(
+              `${bundleDir}/${jsModules}`,
+              Object.entries(self.options.builds).filter(
+                ([ name, options ]) => options.scenes.includes(scene) &&
+                options.outputs.includes('js') &&
+                (!options.condition || options.condition === 'module')
+              ).map(([ name, options ]) => {
+                return `// BUILD: ${name}\n` + fs.readFileSync(`${bundleDir}/${name}-build.js`, 'utf8');
+              }).join('\n')
+            );
+            fs.writeFileSync(
+              `${bundleDir}/${jsNoModules}`,
+              Object.entries(self.options.builds).filter(
+                ([ name, options ]) => options.scenes.includes(scene) &&
+                options.outputs.includes('js') &&
+                (!options.condition || options.condition === 'nomodule')
+              ).map(([ name, options ]) => {
+                return `// BUILD: ${name}\n` + fs.readFileSync(`${bundleDir}/${name}-build.js`, 'utf8');
+              }).join('\n')
+            );
             fs.writeFileSync(
-              `${bundleDir}/${APOS_MERGED_BUNDLE}`,
-              fs.readFileSync(`${bundleDir}/public-bundle.js`) +
-                fs.readFileSync(`${bundleDir}/apos-only-bundle.js`)
+              `${bundleDir}/${css}`,
+              Object.entries(self.options.builds).filter(
+                ([ name, options ]) => options.scenes.includes(scene) &&
+                options.outputs.includes('css')
+              ).map(([ name, options ]) => {
+                return `/* BUILD: ${name} */\n` + fs.readFileSync(`${bundleDir}/${name}-build.css`, 'utf8');
```

Comment:
Being able to figure out where CSS came from (ui/public vs ui/src) can be nice too.